### PR TITLE
Improves spellcaster AI

### PR DESCRIPTION
**Pull request by daedela**

### CODE REVIEW

**Comment by daedela** [comment]
This change adds:
- a new define "IS_SPELLCASTER(ch)" which allows us to easily check if the character is of a spellcasting class.
- stops cleric-type and bard mobs from casting their "hindrance" spells (Blind and Poison mostly) on characters more than 20 levels higher, since the chance of affecting the target is so minimal.  Will cause them to use damaging spells, like Full Harm and Sunray, instead.
- added a new function that will cause cleric-types and bards to run through all their engaged opponents, check if they're a spellcaster, and if so, attempt to cast Insanity or Silence on them.
- re-ordered the list of sorcerer-type offensive spells so they'll correctly cast the more damaging spells first.
-

File: src/cleric.cpp [hunk 1]
```diff
@@ -201,7 +220,7 @@ bool cleric_ai_action(CharData *ch, CharData *victim) {
     }
 
     return false;
-}
+    }
```
Comment (stridera) [inline src/cleric.cpp:223]: Mistake?

**Comment by daedela** [comment]
Oops, yep.

File: include/class.hpp [hunk 2]
```diff
@@ -141,7 +147,7 @@ extern ClassDef classes[NUM_CLASSES];
          ? classes[(int)GET_CLASS(ch)].nowear_flag && OBJ_FLAGGED(obj, classes[(int)GET_CLASS(ch)].nowear_flag)        \
          : 0)
 
-enum level_action { LEVEL_GAIN, LEVEL_LOSE };
+    enum level_action { LEVEL_GAIN, LEVEL_LOSE };
```
Comment (stridera) [inline include/class.hpp:150]: Another mistake.  Looks like your formatter is broken.  It won't break the code, just looks bad.

**Review by stridera — APPROVED** [review]
I'll approve since there doesn't look like any breaking code, but it would look nicer with a formatter running.  :)

File: include/class.hpp (hunk 2 above)
Comment (daedela) [reply include/class.hpp:150]: Yeah not sure what happened there

**Pull request merged**